src/unix.rs

```diff
@@ -61,3 +61,27 @@ pub fn try_wait_without_reaping(handle: Handle) -> io::Result<bool> {
                            format!("unexpected si_signo from waitid: {}", siginfo.si_signo)))
     }
 }
+
+pub fn terminate(child: &super::SharedChild) -> io::Result<()> {
+    child.send_signal(libc::SIGTERM)
+}
+
+pub trait SharedChildExt {
+    /// Sends a signal to the child process with `libc::kill`.
+    fn send_signal(&self, signal: libc::c_int) -> io::Result<()>;
+}
+
+impl SharedChildExt for super::SharedChild {
+    fn send_signal(&self, signal: libc::c_int) -> io::Result<()> {
+        let status = self.state_lock.lock().unwrap();
+        if let super::ChildState::Exited(_) = *status {
+            return Ok(());
```

Comment:
That's an interesting question. If we wanted give the caller as much information as possible about whether the signal "really got sent", how much do we really know from the OS? I'm not familiar enough with the POSIX spec to know what it guarantees. If I do something like `try_wait` + `signal`, I can be sure that "the child is still alive, and this signal is going to get delivered unless it's exiting _right now_." Is that basically the same thing the OS would tell us?